src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryNullCheckWithInstanceOfCheck.java

```diff
@@ -0,0 +1,221 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2025 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import javax.annotation.Nullable;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <div>
+ * Checks for redundant null checks with the instanceof operator.
+ * </div>
+ *
+ * <p>
+ * The instanceof operator inherently returns false when the left operand is null,
+ * making explicit null checks redundant in boolean expressions with instanceof.
+ * </p>
+ *
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ *
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ *
+ * <ul>
+ *     <li>
+ *         {@code unnecessary.nullcheck.with.instanceof}
+ *     </li>
+ * </ul>
+ *
+ * @since 10.24.0
+ */
+@StatelessCheck
+public class UnnecessaryNullCheckWithInstanceOfCheck extends AbstractCheck {
+
+    /**
+     * The error message key for reporting unnecessary null checks.
+     */
+    public static final String MSG_UNNECESSARY_NULLCHECK = "unnecessary.nullcheck.with.instanceof";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.NOT_EQUAL};
```

Comment:
I might missed something, why we listening so frequent token?
It will be more effective to subscribe to LITERAL_INSTANCEOF

It will be called ~hundreds time less..